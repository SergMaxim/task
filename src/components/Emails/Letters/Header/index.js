import React, { Component } from 'react';
import Styles from './styles.scss';
import { func, number } from 'prop-types';

export default class Header extends Component {
    static propTypes = {
        count:         number.isRequired,
        searchLetters: func.isRequired
    };

    constructor () {
        super();
        this.handleSubmit = ::this._handleSubmit;
        this.handleInputChange = ::this._handleInputChange;
    }

    state = {
        comment: ''
    };
    _handleSubmit (event) {
        event.preventDefault();
        const { comment } = this.state;

        this.props.searchLetters(comment);
    }

    _handleInputChange (event) {
        const { value: comment } = event.target;

        this.setState({
            comment
        });
    }
    render () {
        const { count } = this.props;

        return (
            <section className = { Styles.header }>
                <div>
                    Inbox <span>{count}</span>
                </div>
                <form className = { Styles.search } onSubmit = { this.handleSubmit }>
                    <input
                        placeholder = 'search'
                        type = 'text'
                        onKeyUp = { this.handleInputChange }
                    />
                    <input type = 'submit' value = '' />
                </form>
            </section>
        );
    }
}
