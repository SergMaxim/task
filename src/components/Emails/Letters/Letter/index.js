import React from 'react';
import moment from 'moment';
import { bool, number, string } from 'prop-types';

import Styles from './styles.scss';

const Letter = (props) => {
    let style = null;

    if (props.star) {
        style = {
            backgroundPosition: 'top center'
        };
    }

    return (
        <section className = { Styles.letter }>
            <span className = { Styles.star } style = { style } />
            <div className = { Styles.name }> {props.name}</div>
            <div className = { Styles.theme }> {props.theme}</div>

            <time>
                {moment
                    .unix(props.time)
                    .locale('uk')
                    .format('D.M.YY HH:mm')}
            </time>
        </section>
    );
};

Letter.propTypes = {
    name:  string.isRequired,
    star:  bool.isRequired,
    theme: string.isRequired,
    time:  number.isRequired
};

export default Letter;
