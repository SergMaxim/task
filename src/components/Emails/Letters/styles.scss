.letters{
  position: relative;
  border-left: 1px solid #d8e3e7;
  width: 80%;
  height: calc(100vh - 100px);
  background-color: #f5f8f9;

  table{
    width: 100%;
  }
}