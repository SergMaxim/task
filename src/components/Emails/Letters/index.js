import React, { Component } from 'react';
import { arrayOf, bool, shape, string, number, func } from 'prop-types';
import Styles from './styles.scss';
import Letter from './Letter';
import Header from './Header';
import { getUniqueID } from '../../../helpers';

export default class Letters extends Component {
    static propTypes = {
        letters: arrayOf(
            shape({
                name:  string.isRequired,
                star:  bool.isRequired,
                theme: string.isRequired,
                time:  number.isRequired
            }).isRequired
        ).isRequired,
        searchLetters: func.isRequired
    };

    render () {
        const { letters: postsData, searchLetters } = this.props;
        const letters = postsData.map((props) => {
            const id = getUniqueID();

            return <Letter key = { id } { ...props } />;
        });

        return (
            <section className = { Styles.letters }>
                <Header
                    count = { postsData.length }
                    searchLetters = { searchLetters }
                />
                {letters}
            </section>
        );
    }
}
