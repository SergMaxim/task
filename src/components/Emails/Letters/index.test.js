import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import dom from 'react-test-renderer';

// Instruments
import Letters from './';

configure({ adapter: new Adapter() });

const message = 'message';
const props = {
    letters: [
        {
            name:  'name',
            star:  true,
            theme: 'theme',
            time:  1513545795
        },
        {
            name:  'name2',
            star:  false,
            theme: 'theme2',
            time:  1513545795
        }
    ],
    searchLetters: jest.fn()
};
const state = {
    comment: ''
};
const mutatedState = {
    comment: message
};

const result = mount(<Letters { ...props } />);
const renderTree = dom.create(<Letters { ...props } />).toJSON();

describe('Letters component', () => {
    test(`Letters should div 2 'Letter' element`, () => {
        expect(result.find('Letter')).toHaveLength(2);
    });
    test('Letters component should correspond to its snapshot counterpart', () => {
        expect(renderTree).toMatchSnapshot();
    });
});
