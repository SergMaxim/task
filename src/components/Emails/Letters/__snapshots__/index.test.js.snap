// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letters component Letters component should correspond to its snapshot counterpart 1`] = `
<section
  className="letters"
>
  <section
    className="header"
  >
    <div>
      Inbox 
      <span>
        2
      </span>
    </div>
    <form
      className="search"
      onSubmit={[Function]}
    >
      <input
        onKeyUp={[Function]}
        placeholder="search"
        type="text"
      />
      <input
        type="submit"
        value=""
      />
    </form>
  </section>
  <section
    className="letter"
  >
    <span
      className="star"
      style={
        Object {
          "backgroundPosition": "top center",
        }
      }
    />
    <div
      className="name"
    >
       
      name
    </div>
    <div
      className="theme"
    >
       
      theme
    </div>
    <time>
      17.12.17 23:23
    </time>
  </section>
  <section
    className="letter"
  >
    <span
      className="star"
      style={null}
    />
    <div
      className="name"
    >
       
      name2
    </div>
    <div
      className="theme"
    >
       
      theme2
    </div>
    <time>
      17.12.17 23:23
    </time>
  </section>
</section>
`;
